### ISSUE TITLE
Upgrade gaia dependency

### ISSUE THREAD

**Issue by karthik4444** [cosmos/gaia#55]
The following dependency:

```
vendor/github.com/ethanfrey/hid/hid_darwin.go

contains overflow errors. Particularly, the error messages while calling glide install are:

github.com/cosmos/gaia/vendor/github.com/ethanfrey/hid

vendor/github.com/ethanfrey/hid/hid_darwin.go:40: constant 18446744073172681404 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:42: constant 18446744073172681405 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:44: constant 18446744073172681406 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:46: constant 18446744073172681407 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:48: constant 18446744073172681408 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:50: constant 18446744073172681409 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:52: constant 18446744073172681410 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:54: constant 18446744073172681411 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:56: constant 18446744073172681412 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:58: constant 18446744073172681413 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:58: too many errors
? github.com/cosmos/gaia/cmd/gaia	[no test files]
make: *** [test] Error 2
```

**Comment by ethanfrey** [comment]
Okay, I think this was fixed, but the dependency was not updated...

@karthik4444 Can you please check the latest commit master:
`go get -u github.com/ethanfrey/hid`

Let me know if it works for you, and I can update all dependencies.. just waiting for feedback as different macs had different issues.

**Comment by adrianbrink** [comment]
wfm

**Comment by rigelrozanski** [comment]
WFM

**Comment by karthik4444** [comment]
@ethanfrey Still didn't work for me. I had to cast all the consts as uint64 for it to build.

**Comment by jolesbi** [comment]
this is causing an issue for me as well. can't run the cosmos-ui on gaia-1.

```
make install
go install ./cmd/gaia
# github.com/cosmos/gaia/vendor/github.com/ethanfrey/hid
vendor/github.com/ethanfrey/hid/hid_darwin.go:40: constant 18446744073172681404 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:42: constant 18446744073172681405 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:44: constant 18446744073172681406 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:46: constant 18446744073172681407 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:48: constant 18446744073172681408 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:50: constant 18446744073172681409 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:52: constant 18446744073172681410 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:54: constant 18446744073172681411 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:56: constant 18446744073172681412 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:58: constant 18446744073172681413 overflows C.IOReturn
vendor/github.com/ethanfrey/hid/hid_darwin.go:58: too many errors
make: *** [install] Error 2
```

**Comment by karthik4444** [comment]
It seems that this issue is build specific. In some cases, the compiler knows to interpret the constants to uint64 while in my case, you have to cast them specifically.

**Comment by rigelrozanski** [comment]
what version of Go are you using?

**Comment by karthik4444** [comment]
go version go1.9.1 darwin/amd64

**Comment by adrianbrink** [comment]
@karthik4444 This is a temporary fix until the release of `0.4.0`, but here are all binaries. Please check if the binary would work for you.

[gaia-android-16-386.zip](https://github.com/cosmos/gaia/files/1509496/gaia-android-16-386.zip)
[gaia-android-16-arm.zip](https://github.com/cosmos/gaia/files/1509497/gaia-android-16-arm.zip)
[gaia-darwin-10.6-386.zip](https://github.com/cosmos/gaia/files/1509498/gaia-darwin-10.6-386.zip)
[gaia-darwin-10.6-amd64.zip](https://github.com/cosmos/gaia/files/1509499/gaia-darwin-10.6-amd64.zip)
[gaia-linux-386.zip](https://github.com/cosmos/gaia/files/1509500/gaia-linux-386.zip)
[gaia-linux-amd64.zip](https://github.com/cosmos/gaia/files/1509501/gaia-linux-amd64.zip)
[gaia-linux-arm-5.zip](https://github.com/cosmos/gaia/files/1509502/gaia-linux-arm-5.zip)
[gaia-linux-arm-6.zip](https://github.com/cosmos/gaia/files/1509503/gaia-linux-arm-6.zip)
[gaia-linux-arm-7.zip](https://github.com/cosmos/gaia/files/1509504/gaia-linux-arm-7.zip)
[gaia-linux-arm64.zip](https://github.com/cosmos/gaia/files/1509505/gaia-linux-arm64.zip)
[gaia-linux-mips.zip](https://github.com/cosmos/gaia/files/1509506/gaia-linux-mips.zip)
[gaia-linux-mips64.zip](https://github.com/cosmos/gaia/files/1509507/gaia-linux-mips64.zip)
[gaia-linux-mips64le.zip](https://github.com/cosmos/gaia/files/1509508/gaia-linux-mips64le.zip)
[gaia-linux-mipsle.zip](https://github.com/cosmos/gaia/files/1509509/gaia-linux-mipsle.zip)

**Comment by ethanfrey** [comment]
Yes, this is a compiler issue. As of go 1.9, it started to return errors when you cast a int constant to a uint or visa versa (only for constants, not variables), which broke the hid package (I just maintain a fork to have some fixes like this).

It was fixed on master in ethanfrey/hid
The dependencies have been update in cosmos-sdk:develop (8da0f3f)
Gaia developers need to run `glide update` and fix any breaking changes, then it will use the version that compiles properly.

**Comment by rigelrozanski** [comment]
@ethanfrey I needed to roll back a commit that used in a working branch I'm developing - it became dep. hell when you integrated it (and by doing so broke feature/delegation) and I needed to focus on some other fundamental problems.. This issue should still be addressed in this next release

**Comment by ebuchman** [comment]
Seems this might be an issue with go1.9.1 ? jolesbi solved it by upgrading to `1.9.2`. The rest of us on 1.9.2 also don't seem to be having any trouble
